Modules/_testclinic.c

```diff
@@ -0,0 +1,959 @@
+#ifndef Py_BUILD_CORE_BUILTIN
+#  define Py_BUILD_CORE_MODULE 1
+#endif
+
+/* Always enable assertions */
+#undef NDEBUG
+
+#define PY_SSIZE_T_CLEAN
+
+#include "Python.h"
+
+#include "clinic/_testclinic.c.h"
+
+
+/* Pack arguments to a tuple, implicitly increase all the arguments' refcount.
+ * NULL arguments will be replaced to Py_None. */
+static PyObject *
+pack_arguments_newref(int argc, ...)
+{
+    assert(!PyErr_Occurred());
+    PyObject *tuple = PyTuple_New(argc);
+    if (!tuple) {
+        return NULL;
+    }
+
+    va_list vargs;
+    va_start(vargs, argc);
+    for (int i = 0; i < argc; i++) {
+        PyObject *arg = va_arg(vargs, PyObject *);
+        if (arg) {
+            if (_PyObject_IsFreed(arg)) {
+                PyErr_Format(PyExc_AssertionError,
+                             "argument %d at %p is freed or corrupted!",
+                             i, arg);
+                va_end(vargs);
+                Py_DECREF(tuple);
+                return NULL;
+            }
+        }
+        else {
+            arg = Py_None;
+        }
+        PyTuple_SET_ITEM(tuple, i, Py_NewRef(arg));
+    }
+    va_end(vargs);
+    return tuple;
+}
+
+/* Pack arguments to a tuple, all the arguments' refcounts should be increased before passing in.
+ * Do not accept NULL arguments unless error occurs. */
+#define PACK_ARGUMENTS(argc, ...) do { \
+        assert(!PyErr_Occurred()); \
+        PyObject *out[argc] = {NULL,}; \
+        for (int _i = 0; _i < argc; _i++) { \
+            out[_i] = ((PyObject *[]){__VA_ARGS__})[_i]; \
+            assert(out[_i] || PyErr_Occurred()); \
+            if (!out[_i]) { \
+                for (int _j = _i - 1; _j >= 0; _j--) { \
```

Comment:
If `_i` is `0` the API call fails, `_j` will start at `-1`. Better to iterate from zero to `_i`.

```suggestion
                for (int _j = 0; _j >= i; _j++) { \
```